src/__internal__/fieldset/fieldset.component.js

```diff
@@ -5,34 +5,43 @@ import { StyledFieldset, StyledFieldsetContent, StyledLegendContainer } from './
 import ValidationIcon from '../../components/validations/validation-icon.component';
 
 const Fieldset = ({
-  legend, children, inline, error, warning, info, styleOverride, ...rest
-}) => (
-  <StyledFieldset
-    data-component='fieldset'
-    styleOverride={ styleOverride.root }
-    { ...rest }
-  >
-    <StyledFieldsetContent inline={ inline }>
-      {legend && (
-        <StyledLegendContainer
-          inline={ inline }
-          styleOverride={ styleOverride.legend }
-        >
-          <legend>
-            { legend }
-          </legend>
-          <ValidationIcon
-            error={ error }
-            warning={ warning }
-            info={ info }
-            tabIndex={ 0 }
-          />
-        </StyledLegendContainer>
-      )}
-      { children }
-    </StyledFieldsetContent>
-  </StyledFieldset>
-);
+  legend, children, inline, legendWidth, legendAlign, legendSpacing = 2, error,
+  warning, info, ml, styleOverride, ...rest
+}) => {
+  const legendTextAlign = inline ? (legendAlign || 'right') : legendAlign;
```

Comment:
Same as comment above, probably no need for `legendAlign` when component is not `inline`